### ISSUE TITLE
Symbol conflicts when linked statically

### ISSUE THREAD

**Issue by michaelforney** [Yubico/libfido2#75]
Now that OpenSSH is able to link with libfido2 directly to provide built-in security key support, there is the risk of symbol conflicts when it is linked statically:

```
/lib/gcc/x86_64-linux-musl/8.3.0/../../../../x86_64-linux-musl/bin/ld: /home/michael/.local/lib/libfido2.a(log.c.o): in function `log_init':
log.c:(.text+0x0): multiple definition of `log_init'; ./libssh.a(log.o):/home/michael/src/openssh-portable/log.c:258: first defined here
collect2: error: ld returned 1 exit status
```

I'm currently working around this with the following diff:

```diff
diff --git a/src/extern.h b/src/extern.h
index 2f69094..9d322bc 100644
--- a/src/extern.h
+++ b/src/extern.h
@@ -91,6 +91,9 @@ int rx_cbor_status(fido_dev_t *, int);
 #define log_debug(...) do { /* nothing */ } while (0)
 #define log_xxd(...)   do { /* nothing */ } while (0)
 #else
+#define log_init fido_log_init
+#define log_debug fido_log_debug
+#define log_xxd fido_log_xxd
 #ifdef __GNUC__
 void log_init(void);
 void log_debug(const char *, ...) __attribute__((__format__ (printf, 1, 2)));
```

libfido2 should probably use the `fido_` prefix for *all* symbols with external linkage, but there are quite a few others in `extern.h`...

**Comment by martelletto** [comment]
Hi, just a quick note to say thanks for pointing this out and apologies for the breakage. Let me mull this over for a bit and get back to you.

**Comment by martelletto** [comment]
Developed in [prefix_symbols](https://github.com/Yubico/libfido2/tree/prefix_symbols). I tried to be careful and did the conversions by hand, but given the size of the resulting diff, I will revisit it later in the week before merging. Thank you once again for pointing out this issue.

**Comment by martelletto** [comment]
Merged in ad22526c, 32bc199e, 833556b2, 9f76a5c0, 28d54a79, ef45bc47, 7cb6d41d, edf09aea, 05cebe60, and ec7a4531.

**Comment by michaelforney** [comment]
Thanks, your work on this is appreciated!
